rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -258,6 +258,29 @@ inline void microphone_device::variable_byteswap(const void* src, void* dst)
 	}
 }
 
+inline u32 microphone_device::convert_16_bit_pcm_to_float(const std::vector<u8>& buffer, usz num_bytes)
+{
+	static_assert((float_buf_size % sizeof(f32)) == 0);
+
+	float_buf.resize(float_buf_size, 0);
+	const u32 bytes_to_write = static_cast<u32>(num_bytes * (sizeof(f32) / sizeof(s16)));
+	ensure(bytes_to_write <= float_buf.size());
+
+	const be_t<s16>* src = reinterpret_cast<const be_t<s16>*>(static_cast<const void*>(buffer.data()));
```

Comment:
hmm. this still produces a warning on gcc.
maybe try:
```suggestion
	const be_t<s16>* src = utils::bless<const be_t<s16>>(buffer.data());
```